src/main/java/com/example/bookstore/service/impl/BookServiceImpl.java

```diff
@@ -20,20 +21,39 @@ public BookServiceImpl(BookRepository bookRepository, BookMapper bookMapper) {
     }
 
     @Override
-    public BookDto save(CreateBookRequestDto createBookRequestDto) {
-        return bookMapper.toDto(bookRepository.save(bookMapper.toModel(createBookRequestDto)));
+    public BookResponseDto save(BookRequestDto bookRequestDto) {
+        return bookMapper.toDto(bookRepository.save(bookMapper.toModel(bookRequestDto)));
     }
 
     @Override
-    public List<BookDto> getAll() {
-        return bookRepository.getAll().stream()
+    public List<BookResponseDto> getAll() {
+        return bookRepository.findAll().stream()
                 .map(bookMapper::toDto)
                 .toList();
     }
 
     @Override
-    public BookDto getBookById(Long id) {
-        return bookMapper.toDto(bookRepository.findBookById(id)
+    public BookResponseDto getBookById(Long id) {
+        return bookMapper.toDto(bookRepository.findById(id)
                 .orElseThrow(() -> new EntityNotFoundException("Can't find book by id: " + id)));
     }
+
+    @Override
+    public BookResponseDto update(BookRequestDto bookRequestDto, Long id) {
+        if (bookRepository.existsById(id)) {
```

Comment:
Find book by id
use method from mapper bookMapper.updateBookFromDto(bookRequestDto, book);
save book to DB and return dto